### PR TITLE
Refactoring platforms.py into pydantic classes

### PR DESCRIPTION
**Pull request by jlnav**
Note the validators on the base `Platform` class that apply to each of the subclasses. Note how attributes are `.`accessed.

Creating/providing a custom subclass probably needs to be tested

Also, for those functions that access attributes of the classes, maybe a class-instance could be returned to the Resources module instead for direct-use.

For _unknown_ systems, a Platform instance could be constructed on-the-spot with detected values to keep the interface the same.

### CODE REVIEW

File: libensemble/resources/platforms.py [hunk 1]
```diff
@@ -10,91 +14,116 @@
...
-#TODO MAKE ALPHABETICAL
+class Platform(BaseModel):
+    mpi_runner: str = "srun"
+    runner_name: Optional[str]
+    cores_per_node: int
+    logical_cores_per_node: int
+    gpus_per_node: int
+    gpu_setting_type: int
+    gpu_setting_name: str
+    scheduler_match_slots: bool = False
```
Comment (jlnav) [inline libensemble/resources/platforms.py:26]: should default to True

File: libensemble/resources/platforms.py [hunk 2]
```diff
@@ -10,91 +14,116 @@
...
-    "logical_cores_per_node" : 208,  # finds - check
-    "gpus_per_node" : 6,
-    "gpu_setting_type": GPU_SET_DEF,
-    "scheduler_match_slots": True,
-    }
-
 
-#TODO MAKE ALPHABETICAL
+class Platform(BaseModel):
+    mpi_runner: str = "srun"
```
Comment (jlnav) [inline libensemble/resources/platforms.py:19]: should default to mpich

Comment (jlnav) [reply libensemble/resources/platforms.py:19]: nevermind defaults not needed

File: libensemble/resources/platforms.py [hunk 3]
```diff
@@ -10,91 +14,116 @@
...
-    "cores_per_node" : 104,  # finds - check
-    "logical_cores_per_node" : 208,  # finds - check
-    "gpus_per_node" : 6,
-    "gpu_setting_type": GPU_SET_DEF,
-    "scheduler_match_slots": True,
-    }
-
 
-#TODO MAKE ALPHABETICAL
+class Platform(BaseModel):
```
Comment (shuds13) [inline libensemble/resources/platforms.py:18]: We don't want default values.

**Comment by shuds13** [comment]
@jlnav Are most of these commits from a merge?

**Comment by jlnav** [comment]
I believe this is no longer needed to due a manual merge

**Pull request closed without merging**